### PR TITLE
CAL-499 Imaging App threat model

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
Updates and completes the threat model for the Imaging App.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)
@bakejeyner @dcruver @SmithJosh 

#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)
@brjeter
@stustison

#### How should this be tested?
No testing is required short of ensuring that the model can be opened in [Threat Dragon](https://github.com/mike-goodwin/owasp-threat-dragon-desktop) on Windows and Mac (and on Github as well with the webapp if desired).

A screenshot has been provided for a quick once-over but the model should be reviewed in the application for thoroughness.

#### What are the relevant tickets?

[CAL-499](https://codice.atlassian.net/browse/CAL-499)
#### Screenshots (if appropriate)
screen shot 2018-12-03 at 1 17 11 pm

#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: ThreatDragonModels/ImagingApp/ImagingApp.json [hunk 1]
```diff
@@ -37,14 +37,242 @@
...
+                  "type": "Denial of service",
+                  "title": "Potential Process Crash Caused by System Overload",
+                  "description": "A large number of NITFs or a smaller number of very large NITFs can be uploaded to be processed concurrently, using all available server heap. Consider out-of-band processing, stream-based processing, and queueing of images to be processed in an on-disk holding area.",
+                  "$$hashKey": "object:497"
+                },
+                {
+                  "status": "Open",
+                  "severity": "High",
+                  "type": "Denial of service",
+                  "title": "Malicious NITF causes process/stack overflow",
```
Comment (brjeter) [inline ThreatDragonModels/ImagingApp/ImagingApp.json]: ❓ Could we say that this threat is mitigated if the imaging code does not allow for such a scenario?

File: ThreatDragonModels/ImagingApp/ImagingApp.json [hunk 2]
```diff
@@ -37,14 +37,242 @@
...
+                  "description": "It is theoretically possible to craft a NITF in such a way that it can cause memory or stack overflow issues.\n\nThe likelihood of this is low or nil and no example of such a NITF is known at this time.",
```
Comment (brjeter) [inline ThreatDragonModels/ImagingApp/ImagingApp.json]: ✏️ It seems that the second sentence about the likelihood either
1. doesn't belong or
2. should be an explanation for the "mitigation", if the answer to my above question is yes.

File: ThreatDragonModels/ImagingApp/ImagingApp.json [hunk 3]
```diff
@@ -37,14 +37,242 @@
...
+                  "$$hashKey": "object:531"
+                },
+                {
+                  "status": "Open",
+                  "severity": "High",
+                  "title": "Malicious NITF triggers execution of system code",
```
Comment (brjeter) [inline ThreatDragonModels/ImagingApp/ImagingApp.json]: ❓ Could we say that this threat is mitigated if the imaging code does not allow for such a scenario?

File: ThreatDragonModels/ImagingApp/ImagingApp.json [hunk 4]
```diff
@@ -37,14 +37,242 @@
...
+                  "type": "Elevation of privilege",
+                  "description": "It is theoretically possible to craft a NITF in such a way that it can cause third-party image processing libraries to execute arbitrary system code.\n\nThe likelihood of this is low or nil and no example of such a NITF is known at this time.",
```
Comment (brjeter) [inline ThreatDragonModels/ImagingApp/ImagingApp.json]: ✏️ It seems that the second sentence about the likelihood either
1. doesn't belong or
2. should be an explanation for the "mitigation", if the answer to my above question is yes.

File: ThreatDragonModels/ImagingApp/ImagingApp.json [hunk 5]
```diff
@@ -37,14 +37,242 @@
...
+                  "$$hashKey": "object:525"
+                },
+                {
+                  "status": "Mitigated",
+                  "severity": "Medium",
+                  "type": "Repudiation",
+                  "title": "NITF source claims Image App did not process image",
+                  "description": "NITF source (file system, user at Karaf console, user of UI, etc.) claims an image was not processed as requested.",
```
Comment (SmithJosh) [inline ThreatDragonModels/ImagingApp/ImagingApp.json]: ❓ Is this really a threat? I was imagining the repudiation scenario as the NITF source claiming they did not upload a particular (potentially malicious) NITF.

Comment (coyotesqrl) [reply ThreatDragonModels/ImagingApp/ImagingApp.json]: Good point. Might need to add a second repudiation threat.

**Comment by coyotesqrl** [comment]
build now

**Comment by cxbot1** [comment]
Internal build has been scheduled, your results will be available at build completion.

**Comment by cxbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.codice.org/job/Alliance-Jobs/job/pr/job/Linux/1048/
:white_check_mark: JOB SUCCESS

**Comment by dcruver** [comment]
NITFs are basically binary image data plus text metadata. When we ingest, we write the metadata to SOLR and the image data to the `StorageProvider` which writes that to disk. So I think we should consider separating the DDF Catalog Framework bubble out into CatalogProvider/StorageProvider to show how we handle the two parts of the NITF.

**Comment by coyotesqrl** [comment]
Updated per @dcruver and @SmithJosh comments.

screen shot 2018-12-05 at 7 07 46 am

**Comment by coyotesqrl** [comment]
build now

**Comment by cxbot1** [comment]
Internal build has been scheduled, your results will be available at build completion.

**Comment by coyotesqrl** [comment]
Jira ticket created to investigate two of the unmitigated threats: https://codice.atlassian.net/browse/CAL-501

**Comment by cxbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.codice.org/job/Alliance-Jobs/job/pr/job/Linux/1049/
:white_check_mark: JOB SUCCESS

**Pull request merged**